native-sql-engine/core/src/main/scala/org/apache/spark/sql/execution/ColumnarBroadcastExchangeExec.scala

```diff
@@ -82,11 +84,23 @@ case class ColumnarBroadcastExchangeExec(mode: BroadcastMode, child: SparkPlan)
   @transient
   lazy val completionFuture: scala.concurrent.Future[broadcast.Broadcast[Any]] =
     promise.future
-
+  
+  @transient
+  private lazy val maxBroadcastRows = mode match {
```

Comment:
Fixed through shim layer. 
TDOO: whether the update on spark 3.2 is applicable to 3.1.